x/mev/module.go

```diff
@@ -0,0 +1,203 @@
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return nil
+	//return cli.GetQueryCmd()
+}
+
+type AppModule struct {
+	AppModuleBasic
+	keeper *mev.Keeper
+}
+
+func NewAppModule(cdc codec.Codec, k *mev.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: NewAppModuleBasic(cdc),
+		keeper:         k,
+	}
+}
+
+// Name returns the mev module's name.
+func (AppModule) Name() string {
+	return types.ModuleName
+}
+
+// RegisterInvariants registers the mev module invariants.
+func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+
+func NewHandler(k *mev.Keeper) sdk.Handler {
+	//msgServer := keeper.NewMsgServerImpl(k)
+
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+
+		fmt.Println("MEV HANDLER MEV")
+
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+
+		switch msg := msg.(type) {
+		//case *types.MsgEVMTransaction:
+		//	res, err := msgServer.EVMTransaction(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		//case *types.MsgSend:
+		//	res, err := msgServer.Send(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		//case *types.MsgRegisterPointer:
+		//	res, err := msgServer.RegisterPointer(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		//case *types.MsgAssociateContractAddress:
+		//	res, err := msgServer.AssociateContractAddress(sdk.WrapSDKContext(ctx), msg)
+		//	return sdk.WrapServiceResult(ctx, res, err)
+		default:
+			errMsg := fmt.Sprintf("unrecognized %s message type MEV MEV MEV: %T", types.ModuleName, msg)
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+		}
+	}
+}
+
+// Route returns the message routing key for the mev module.
+func (am AppModule) Route() sdk.Route {
+	return sdk.NewRoute(types.RouterKey, NewHandler(am.keeper))
+}
+
+// QuerierRoute returns the mev module's querier route name.
+func (AppModule) QuerierRoute() string {
+	return types.QuerierRoute
+}
+
+// LegacyQuerierHandler returns the mev module sdk.Querier.
+func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
+	return nil
+}
+
+// InitGenesis performs genesis initialization for the mev module.
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
+	var genesisState types.GenesisState
+	cdc.MustUnmarshalJSON(data, &genesisState)
+	return []abci.ValidatorUpdate{}
+}
+
+// ExportGenesis returns the exported genesis state as raw bytes for the mev module.
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
+	return cdc.MustMarshalJSON(types.DefaultGenesis())
+}
+
+// ExportGenesisStream returns the mev module's exported genesis state as raw JSON bytes in a streaming fashion.
+func (am AppModule) ExportGenesisStream(ctx sdk.Context, cdc codec.JSONCodec) <-chan json.RawMessage {
+	ch := make(chan json.RawMessage)
+	go func() {
+		ch <- am.ExportGenesis(ctx, cdc)
```

Comment:
send with a select against the ctx to avoid a deadlock.

Reply:
As above, boilerplate and how most modules do this. But I wonder how would this possibly deadlock? Since it runs in a goroutine it should wait until somebody reads it